packages/backend/src/plugins/scaffolder.ts

```diff
@@ -43,7 +44,8 @@ export default async function createPlugin({
   const dockerClient = new Docker();
 
   const discovery = SingleHostDiscovery.fromConfig(config);
-  const entityClient = new CatalogEntityClient({ discovery });
+  const catalogClient = new CatalogClient({ discoveryApi: discovery });
+  const entityClient = new CatalogEntityClient({ catalogClient });
```

Comment:
This modelling is nicer but I think it would be better to treat this as a separate deliberate change, perhaps establishing a pattern of using a common catalog client for all backend plugins so that it's easier to use custom catalog implementations.

Right now I think it's a bit unnecessary as it's a pretty impactful breaking change that requires manual updating in all apps, and I'm not sure it's worth the hassle just yet.

So what I'm suggesting is to just keep passing in the discovery instance and construct the catalog client inside the catalog entity client for now

Reply:
Not sure I follow - the current interface is to pass in a CatalogEntityClient. Wouldn't changing that to a Discovery instance also break all apps? I can make the change, I just want to verify that I understood what you prefer.

Reply:
I am a muppet, ignore the question ☝️